lib/fetch/headers.js

```diff
@@ -16,6 +16,21 @@ const assert = require('assert')
 const kHeadersMap = Symbol('headers map')
 const kHeadersSortedMap = Symbol('headers map sorted')
 
+/**
+ * @param {number} code
+ */
+function isHTTPWhiteSpaceCharCode (code) {
+  switch (code) {
+    case 0x00a:
+    case 0x00d:
+    case 0x009:
+    case 0x020:
+      return true
+    default:
+      return false
+  }
+}
```

Comment:
If we are going to micro optimize I recommend giving the optimizer a break. Something like this should be a single expression.

```suggestion
function isHTTPWhiteSpaceCharCode (code) {
  return code == 0x00a || code == 0x00d || code == 0x009 || code == 0x020;
}
```

Reply:
Hello!
How about this way?
```js
const kIsHTTPWhitespaceBit = (1 << ' '.charCodeAt(0)) |
(1 << '\n'.charCodeAt(0)) |
(1 << '\r'.charCodeAt(0)) |
(1 << '\t'.charCodeAt(0))

/**
 * @see https://fetch.spec.whatwg.org/#concept-header-value-normalize
 * @param {string} potentialValue
 */
function headerValueNormalize (potentialValue) {
  //  To normalize a byte sequence potentialValue, remove
  //  any leading and trailing HTTP whitespace bytes from
  //  potentialValue.
  let i = 0
  let l = potentialValue.length

  while (l > i && ((1 << potentialValue.charCodeAt(l - 1)) & kIsHTTPWhitespaceBit) !== 0) --l
  while (i > l && ((1 << potentialValue.charCodeAt(i)) & kIsHTTPWhitespaceBit) !== 0) ++i

  return i === 0 && l === potentialValue.length ? potentialValue : potentialValue.substring(i, l)
}
```

Reply:
Thanks for letting me know.